### PR TITLE
Documentation: Update CONTRIBUTING docs

### PR DESCRIPTION
**Pull request by gziolo**
## Description

This PR seeks to make it easier to discover developer related documentation explaining how to contribute. It essentially extracts more detailed sections from the top level `CONTRIBUTING.md` file to their own files. The side effect of this change is that they will be also exposed in Gutenberg Handbook in addition to GitHub repository.

### CODE REVIEW

**Review by draganescu — APPROVED** [review]
I think this is a step in the right direction for updating the contributing documentation structure. In the current file the length itself is a problem. where stopping at a certain point would require a lot of "scanning" to get back in the flow.

With this new structure two things are achieved:

1. There is an easier way to follow instructions and read through them
2. The information is easier to refer to, packed in individual smaller bits

**Comment by gziolo** [comment]
@draganescu - thank you for confirming that this is the change in the right direction. I appreciate your input 👍

File: docs/contributors/develop.md [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 
-Please see [CONTRIBUTING.md](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md) for technical details on how to setup and make contributions to the Gutenberg repository.
+Please also see [CONTRIBUTING.md](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md) for general information about contributions to the Gutenberg repository.
 
 The following resources offer additional information for developers who wish to contribute to Gutenberg:
 
+* [Getting Started](/docs/contributors/getting-started.md).
+* [Git Workflow](/docs/contributors/git-workflow.md).
 * [Coding Guidelines](/docs/contributors/coding-guidelines.md) outline additional patterns and conventions used in the Gutenberg project.
-* [Gutenberg Block Grammar](/docs/contributors/grammar.md)
```
Comment (aduth) [inline docs/contributors/develop.md]: I don't know if it was intentional, but I like the reordering to have the more advanced topics arranged further down the list.

**Review by aduth — COMMENTED** [review]
Nice improvements here. 👍  I also think by moving the bulk of the developer-oriented documentation out of `CONTRIBUTING.md`, we can more easily introduce tips for a variety of contribution types (already design and documentation) as being of equal emphasis.

File: docs/contributors/develop.md (hunk 1 above)
Comment (gziolo) [reply docs/contributors/develop.md]: Yes, my idea was to put at the top the most useful documents in the context of new contributions.

**Pull request merged**